configure.ac

```diff
@@ -383,52 +410,88 @@ if test "x$build_dtls" = "xyes"; then
-    # we try first GnuTLS for usability ...
-    if test "x$have_gnutls" = "xyes" -a "x$with_gnutls" = "xno" -a "x$with_tinydtls" = "xno"; then
-        gnutls_version=`pkg-config --modversion gnutls`
-        AX_CHECK_GNUTLS_VERSION
-        AC_MSG_NOTICE([Using auto selected library GnuTLS for DTLS support!])
-        with_gnutls_auto="yes"
-        have_openssl="no" # don't confuse AC_MSG_RESULT at the end of the script
-    fi
-
-    # ... and if not found check OpenSSL is suitable.
-    if test "x$have_openssl" = "xyes" -a "x$with_openssl" = "xno" -a "x$with_gnutls_auto" = "x" -a "x$with_tinydtls" = "xno"; then
-        openssl_version=`pkg-config --modversion openssl`
-        AX_CHECK_OPENSSL_VERSION
-        AC_MSG_NOTICE([Using auto selected library OpenSSL for DTLS support!])
-        with_openssl_auto="yes"
-        have_gnutls="no" # don't confuse AC_MSG_RESULT at the end of the script
-    fi
-
-    # Note that tinyDTLS is used only when explicitly requested.
-
-    # Giving out an error message if we haven't found at least one crypto library.
-    if test "x$have_gnutls" = "xno" -a "x$have_openssl" = "xno" -a "x$have_tinydtls" = "xno"; then
-        AC_MSG_ERROR([==> Option '--enable-dtls' is set but one of the needed cryptography library GnuTLS nor OpenSSL nor tinyDTLS could be found!
-                      Install at least one of the package(s) that contains the development files for GnuTLS (>= $gnutls_version_required) or OpenSSL(>= $openssl_version_required)
-                      or disable the DTLS support using '--disable-dtls'.])
+    if test "$TLSCOUNT" -eq 0; then
+      # The user hasn't requested the use of a specific cryptography library
+      # we try first GnuTLS for usability ...
+      if test "x$have_gnutls" = "xyes"; then
+          gnutls_version=`pkg-config --modversion gnutls`
+          AX_CHECK_GNUTLS_VERSION
+          AC_MSG_NOTICE([Using auto selected library GnuTLS for DTLS support!])
+          with_gnutls_auto="yes"
+          have_openssl="no" # don't confuse AC_MSG_RESULT at the end of the script
+          have_mbedtls="no" # don't confuse AC_MSG_RESULT at the end of the script
+          have_tinydtls="no" # don't confuse AC_MSG_RESULT at the end of the script
+
+      # ... and if not found check OpenSSL is suitable.
+      elif test "x$have_openssl" = "xyes"; then
+          openssl_version=`pkg-config --modversion openssl`
+          AX_CHECK_OPENSSL_VERSION
+          AC_MSG_NOTICE([Using auto selected library OpenSSL for DTLS support!])
+          with_openssl_auto="yes"
+          have_gnutls="no" # don't confuse AC_MSG_RESULT at the end of the script
+          have_mbedtls="no" # don't confuse AC_MSG_RESULT at the end of the script
+          have_tinydtls="no" # don't confuse AC_MSG_RESULT at the end of the script
+
+      # ... and if not found check MbedTLS is suitable.
+      elif test "x$have_mbedtls" = "xyes"; then
+          mbedtls_version=`pkg-config --modversion mbedtls`
```

Comment:
Just a side note: Apparently, there is currently no mbedtls.pc (see ARMmbed/mbedtls#1753). I do not think this is an issue we need to address here but this leaves `mbedtls_version` empty, destroying the final configure output slightly:

```
      build DTLS support      : "yes"
         -->  MbedTLS around  : "yes" (found MbedTLS )
              MBEDTLS_CFLAGS  : ""
              MBEDTLS_LIBS    : "-lmbedtls -lmbedcrypto -lmbedx509"
```

Reply:
Removed  $mbedtls_version for now and disabled `mbedtls_version=`pkg-config --modversion mbedtls`` test